File: oops/polymorphism.py
#more than one form or many form
#different behave

#1)method overloading
#2)method overriding
#3)operator overloading

##method overloading
#same method different number of arguments

class Maths:
    def add(self):
        print("inside no arg")

    def add(self,num1):
        print("inside one arg")

    def add(self,num1,num2):
        print("inside two arg")

m=Maths()
m.add(1,2)

#drawback ==only recently call method is perform

File: javascript/polymorphism.js
//many forms
//method overloading==many forms different number of arguments
//method overriding

//method overloading eg:
class Calc{
    add(){
        console.log("onside no arg method");
    }
    add(num){
        console.log("inside one arg");
    }
    add(num1,num2){
        console.log("inside two arg ");
    }
}

var Cal=new  Calc()
Cal.add(10,20);
Cal.add(10);
